File: solutions/python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit, factors):

    m = []

    for factor in factors:
        if factor != 0:
            for i in range(1, limit // factor + 1):
                if factor * i < limit:
                    m.append(factor * i)
                else:
                    break
        else:
            m.append(0)

    if m == []:
        return 0

    return sum(set(m))

File: solutions/julia/sum-of-multiples/sum-of-multiples.jl
function sum_of_multiples(limit, factors)

    m = []

    for factor in factors
        if factor != 0
            for i in 1 : typemax(Int)
                if factor * i < limit
                    append!(m, factor * i)
                else
                    break
                end
            end
        else
            append!(m, 0)
        end

    end

    if m == [] return 0 end

    return sum(Set(m))
end
